### PR TITLE
improve the fallbacks for opening tabs with hint

### PR DESCRIPTION
**Pull request by ThunderKey**
Firefox does not allow browser.tabs.create to create new tabs with file:// urls (see [MSDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/create#url)).

The current fallback just clicked the links and therefore replaced the current page.

The new workaround adds the appropriate modifier keys to the click. This allows the links to still be opened in tabs, either in the background or foreground.

### CODE REVIEW

**Comment by bovine3dom** [comment]
Hey, thanks for this PR. I think it would be better to add the desired target (current tab, new tab, new window) to the simulateClick function here https://github.com/tridactyl/tridactyl/blob/2f1f6520be5f93c855d0250b1f19e0733af5bf2f/src/lib/dom.ts#L672

That way we could preserve the workarounds for _blank and _new links which would otherwise be blocked by the popup blocker.

**Comment by ThunderKey** [comment]
Thanks for the feedback. Adding options to the `simulateClick` is a good idea.

The issue with the "_blank" and "_new" workarounds is, that it uses `openInNewTab` which will fail in the case of file: urls.

I could therefore add it to the `simulateClick` function, but skip the "_blank" and "_new" workarounds in those cases and pass additional options to the `mouseEvent`, but I'm not sure if this makes the code very clear.

**Comment by bovine3dom** [comment]
Yeah just add a case `&& !( [file url])` here https://github.com/tridactyl/tridactyl/blob/2f1f6520be5f93c855d0250b1f19e0733af5bf2f/src/lib/dom.ts#L680 maybe with a comment that says file URLs aren't supported by openInNewTab

File: src/lib/dom.ts [hunk 1]
```diff
@@ -668,34 +668,61 @@ export function anchors(includeInvisible = false) {
...
     // magic 'short lived event handler' context.
     //
     // OTOH, hardly anyone uses that functionality any more.
-    if (
+    let usePopupBlockerWorkaround =
         (target as HTMLAnchorElement).target === "_blank" ||
         (target as HTMLAnchorElement).target === "_new"
-    ) {
+    const href = (target as HTMLAnchorElement).href
+    if (href !== undefined && href.startsWith("file:")) {
```
Comment (bovine3dom) [inline src/lib/dom.ts:699]: You can simplify this to

```suggestion
    if (href?.startsWith("file:")) {
```

Comment (ThunderKey) [reply src/lib/dom.ts]: Thanks for the tip :)

**Comment by bovine3dom** [comment]
LGTM, thanks!

**Pull request merged**